### DISCUSSION TITLE
Comparing Trees

### DISCUSSION THREAD

**Discussion by mkeshav** [scalaz/scalaz#2300] [Q&A]
Hi,
   Sorry if this is a basic question. I was not able to find documentation or examples of comparing 2 trees

For example, 

    def t1: Tree[Char] =
      'P'.node(
        'O'.node(
          'L'.node('N'.leaf, 'T'.leaf),
          'Y'.node('S'.leaf, 'A'.leaf)),
        'L'.node(
          'W'.node('C'.leaf, 'R'.leaf),
          'A'.node('A'.leaf, 'C'.leaf)))

    def t2: Tree[Char] =
      'P'.node(
        'O'.node(
          'L'.node('N'.leaf, 'T'.leaf),
          'Y'.node('S'.leaf, 'A'.leaf)),
        'L'.node(
          'W'.node('C'.leaf, 'R'.leaf),
          'A'.node('A'.leaf, 'C'.leaf)))

t1 == t2 is false

please point me to the doco or test from which i can learn how to compare trees

Thanks in advance

**Comment by tonymorris** [comment]
I think you want the `Equal` type-class
